docs/messaging/messaging-spans.md

```diff
@@ -104,6 +106,23 @@ First, the lower-level receiving of a message at a consumer, and then the logica
 Often, the waiting for a message is not particularly interesting and hidden away in a framework that only invokes some handler function to process a message once one is received
 (in the same way that the listening on a TCP port for an incoming HTTP message is not particularly interesting).
 
+#### Consumer groups
+
+Consumer groups provide a logical grouping for the message consumers. Messaging systems use them to
+load balance message consumption within the group, broadcast messages to multiple types of the consumers,
+manage offset for each group independently.
+As a result, different groups of consumers can receive messages at a different pace
+or using different settings.
+
+#### Subscriptions
+
+Subscriptions represent entities within messaging systems that allow multiple consumers
+to receive messages from the topic following subscription-specific consumption behavior that includes
+load balancing, durability, filtering, or other system-specific capabilities.
+
+Named subscriptions and consumers groups are semantically different mechanisms messaging systems use
+for similar scenarios such as load-balancing or broadcasting.
```

Comment:
Do you think it would be safe to say that consumer groups apply to checkpoint-based settlement (you have a "topic"), whereas with subscriptions you have message-based settlement (you have a "queue")?

Reply:
Subscription names always apply to topics (fan out doesn't work with queues).

RocketMQ has topics, queues, per-message acks AND consumer groups - https://rocketmq.apache.org/docs/domainModel/03messagequeue

So it's a wild west.

Reply:
I understand, delineations aren't that clear.